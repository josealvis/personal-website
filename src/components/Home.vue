<template>
  <main>
    <section id="intro">
      <div class="center-box-container">
        <h1 id="title">Jose Alvis Pena</h1>
        <span id="subtitle">Fullstack web developer</span>
        <label class="hidden-accesibility" for="input-name"
          >Type your name and press "ENTER"</label
        >
        <input
          id="input-name"
          type="text"
          name="name"
          v-model="visiterName"
          placeholder='Type your name and press "ENTER".'
          v-on:keyup.enter="goTo('about-me')"
          autofocus
        />
      </div>
    </section>
    <section id="about-me">
      <div class="center-box-container">
        <h1>Who I am?</h1>
        <p>
          {{ visiterName ? `Hi ${capitalizeName(visiterName)}.`: null}} It is nice to meet you. let me introduce myself, my name is
          Jose Alvis I am a fullstack web developer. I have flew years working
          in this carrer, more than five if I have to tell. Since that I have
          been Developing web applications for diferent companies, working with
          Git flow, DevOps in SCRUM and AGILE methodologies. My main skill set
          are <span class="hightlight">Javascript ReactJs</span>, C# MVC .net
          core and SQL server, but I also work with other langujes and
          frameworks like Vuejs, NodeJs, Python. Also I have to mantion “I relly
          love to learn new skills”.
        </p>
        <a id="about-link" class="yellow-link" target="_blank" href="http://www.linkedin.com/in/jose-alvis-peña-bonnet-b0194820">
        If you want to know more about my profsional carrear  please visit my 
        <font-awesome-icon :icon="['fab', 'linkedin']" />.
        </a>
      </div>
    </section>
  </main>
</template>

<script>

export default {
  name: "Home",
  data(){
    return {
      visiterName: null
    }
  },
   methods: {
      capitalizeName(fullName) {
        let names =  fullName.match(/\w+/g)
        return names.map(name=>name[0].toUpperCase() + name.slice(1).toLowerCase()).join(" ");
      },
      goTo(section){
          window.location = '#'+section;
      }
    }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
main {
  /* padding-top: 60px; */
  display: flex;
  flex-direction: column;
}
#title {
  margin-top: 20%;
  font-family: "Century Gothic", "Times New Roman", Times, serif;
  font-size: 6em;
  font-weight: bold;
  letter-spacing: 0px;
}

section {
  display: flex;
  flex-direction: column;
  height: 100vh;
  min-height: 500px;
  align-items: center;
}

section h1 {
  align-self: flex-start;
}
#intro {
  align-items: center;
}

.center-box-container {
  display: flex;
  flex-direction: column;
  width: 750px;
  margin-top: 100px;
}

#subtitle {
  align-self: flex-end;
  position: relative;
  top: -95px;
  font-size: 1.5em;
  margin-right: 20px;
  font-weight: bold;
}

#input-name {
  position: relative;
  top: -50px;
  border: none;
  align-self: flex-start;
  padding: 20px;
  min-height: 25px;
  min-width: 400px;
  font-size: 1.2em;
}

#input-name:focus {
  outline: none;
}

#about-me {
  align-items: center;
  background-color: #0a1128;
  color: #fefcfb;
}

.hidden-accesibility {
  position: absolute;
  left: -10000px;
  width: 1px;
  height: 1px;
  top: auto;
  overflow: hidden;
}

.hightlight {
  color: #ecac00;
}

#about-link{
 margin-top: 50px;
}

.yellow-link{
  color: #ecac00;
  text-decoration: none;
}

.yellow-link:hover{
  color: #ecac00;
  text-decoration: underline;
}
</style>
