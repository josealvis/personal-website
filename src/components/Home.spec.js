import { mount } from '@vue/test-utils'
import Home from './Home.vue'

it('renders correctly', () => {
    const wrapper = mount(Home)
    expect(wrapper.element).toMatchSnapshot();
  });