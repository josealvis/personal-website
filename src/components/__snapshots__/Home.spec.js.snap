// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main>
  
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  
  <section
    data-test="intro"
    id="intro"
  >
    <div
      class="center-box-container"
    >
      <h1
        id="title"
      >
        Jose Alvis Pena
      </h1>
      <span
        id="subtitle"
      >
        Fullstack web developer
      </span>
      <label
        class="hidden-accesibility"
        for="input-name"
      >
        Type your name and press "ENTER"
      </label>
      <form
        name="form-vister-name"
      >
        <input
          autofocus=""
          id="input-name"
          name="name"
          placeholder="Type your name and press \\"ENTER\\"."
          type="text"
        />
      </form>
    </div>
  </section>
  <section
    id="about-me"
  >
    <div
      class="center-box-container"
    >
      <h1>
        Who I am?
      </h1>
      <p
        data-test="about-text"
      >
         It is nice to meet you. let me introduce myself, my name is Jose Alvis I am a fullstack web developer. I have flew years working in this carrer, more than five if I have to tell. Since that I have been developing web applications for diferent companies, working with Git flow, DevOps in SCRUM and AGILE methodologies. My main skill set are 
        <span
          class="hightlight"
        >
          Javascript
        </span>
        , 
        <span
          class="hightlight"
        >
          ReactJs
        </span>
        , 
        <span
          class="hightlight"
        >
          C# MVC .net core
        </span>
         and 
        <span
          class="hightlight"
        >
          SQL server
        </span>
        , but I also work with other languages and frameworks like 
        <span
          class="hightlight"
        >
          Vuejs
        </span>
        , 
        <span
          class="hightlight"
        >
          NodeJs
        </span>
        , 
        <span
          class="hightlight"
        >
          Python
        </span>
        . Also I have to mantion “I relly love to learn new skills”. 
      </p>
      <a
        class="yellow-link"
        href="http://www.linkedin.com/in/jose-alvis-peña-bonnet-b0194820"
        id="about-link"
        target="_blank"
      >
         If you want to know more about my career please visit my 
        <font-awesome-icon
          icon="fab,linkedin"
        />
        . 
      </a>
    </div>
  </section>
  <section
    id="portfolio"
  >
    <h1
      class="centered"
      data-test="test-title"
    >
      Here are some project and programming problem I have done 
    </h1>
    <span
      id="portfolio-subtitle"
    >
      Just to take an idea of how I code.
    </span>
    <div
      id="container-portfolio-display"
    >
      <a
        alt="repo of this website"
        class="project-box"
        href="https://github.com/josealvis/personal-website"
        target="_blanck"
      >
        <img
          alt="repo programming challenges"
          class="responsive-img"
          src="[object Object]"
        />
        <div
          class="project-footer"
        >
          <span
            class="project-box-label"
          >
            &lt; This Page Code /&gt;
          </span>
        </div>
      </a>
      <a
        alt="repo programming challenges"
        class="project-box"
        href="https://github.com/josealvis/MayleetCodeChallenge"
        target="_blanck"
      >
        <img
          alt="repo programming challenges"
          class="responsive-img"
          src="[object Object]"
        />
        <div
          class="project-footer"
        >
          <span
            class="project-box-label"
          >
            &lt; Progamming Problems /&gt;
          </span>
        </div>
      </a>
    </div>
  </section>
  <section
    id="readings"
  >
    <div
      class="center-box-container"
    >
      <h1
        class="centered"
      >
        What I am reading now
      </h1>
      <div
        id="container-books-display"
      >
        <div
          class="imag-box"
        >
          <a
            alt="EloquentJs webpage"
            href="https://eloquentjavascript.net/"
            target="_black"
          >
            <img
              alt="art book of EloquentJs"
              class="responsive-img"
              src="https://eloquentjavascript.net/img/cover.jpg"
            />
          </a>
        </div>
      </div>
    </div>
  </section>
  <section
    id="contact"
  >
    <div
      class="center-box-container"
    >
      <h1
        class="centered"
        data-test="contact-title"
      >
        Let’s work together... 
      </h1>
      <span
        class="centered"
        id="contact-subtitle"
      >
        contact me at
      </span>
      <div
        id="social-media-bar"
      >
        <a
          href="https://www.linkedin.com/in/jose-alvis-peña-bonnet-b0194820"
          target="_blanck"
          title="linkedIn profile"
        >
          <font-awesome-icon
            icon="fab,linkedin"
          />
        </a>
        <a
          href="https://github.com/josealvis"
          target="_blanck"
          title="github profile"
        >
          <font-awesome-icon
            icon="fab,github"
          />
        </a>
        <a
          href="mailto: jalvis9114@gmail.com"
          title="Send me a email"
        >
          <font-awesome-icon
            icon="fas,envelope"
          />
        </a>
      </div>
    </div>
  </section>
  <footer>
    <p>
      © 2021 Jose Alvis Pena. Made with 
      <a
        class="footer-link"
        href="https://vuejs.org/"
      >
        Vue.js
      </a>
       + CSS
    </p>
  </footer>
</main>
`;
